DEVI-859 - Webhook Safelist IP changes

mitchellmc opened the pull request:
## Description

 - Add a page with the new EIP/NAT Gateway Fixed IPs that squids will use to deliver Generic Webhooks
 - requires https://github.com/PagerDuty/developer-site/pull/642

## Jira Ticket

 - [DEVI-859](https://pagerduty.atlassian.net/browse/DEVI-859)

## Before Merging!

 - [ ] Check [staging environment](https://developer-v2.pd-staging.com/docs) to ensure changes look as intended.
 - [ ] Ensure there is a review from DevFoundations and from Community.
![6aptyu](https://user-images.githubusercontent.com/4094538/160871184-e5ae781a-0545-4952-9c19-029b09224f21.jpg)

ChezCrawford commented on docs/webhooks/11-Safelist-IPs.md:
```diff
@@ -0,0 +1,36 @@
+If you are consuming Generic Webhooks from PagerDuty, use these IPs to make sure make sure they originate from a known PagerDuty Safelist IP.
+
+
+The full list of IP addresses that our webhook notifications may come from is:
```
Does this list include the current IP addresses or only the static addresses that will be active after the cutover?

ChezCrawford replied:
If the latter, we likely want to include some language that describes the situation: 

* The `/webhook_ips` is the authoritative source until 'x' date.
* This list is the authoritative source after 'x' date.
* If you want total continuity in your firewall, you will need to safelist both sets of ip addresses some time before 'x' date.

mitchellmc replied:
Complicated - this list will have _some_ of the IPS from `/webhook_ips` because the OWT NAT Gateway list IPs are in there. Maybe we should just put that language in anyways?

ChezCrawford commented on toc.json:
```diff
@@ -288,4 +293,4 @@
     }
   ]
-}
+}
```
nit: leave newline at EOF.

mitchellmc replied:
yep

ChezCrawford commented on docs/webhooks/11-Safelist-IPs.md:
```diff
@@ -0,0 +1,36 @@
+
+The full list of IP addresses that our webhook notifications may come from is:
+
+    44.242.69.192
```
Do we want to split these up by service regions in order to match the fact that there are two separate mTLS certificates?

mitchellmc replied:
So - I definitely do not think we should split them up by AWS Region. It's a problem because we deliver some outbound payloads for US customers through EU AWS regions and the list will look confusing for sure. I didn't think there was a good reason to let that complexity leak outwards.

We *can* split them up by Customer region (US vs EU), but there will be some duplicated IPs between those two.

ChezCrawford replied:
> It's a problem because we deliver some outbound payloads for US customers through EU AWS regions 

For the EU OWTs that are connected to the US environment, are those also used by our EU environment?

ChezCrawford commented on docs/webhooks/11-Safelist-IPs.md:
```diff
@@ -0,0 +1,36 @@
+    35.86.21.47
+    52.88.94.18
+    44.238.89.29
+    54.241.68.46
```
Does this IP list include the EIPs used by the US OWTs?

mitchellmc replied:
It should/will as soon as I double check the EU configs and reformat the list to be split by customer region

ChezCrawford commented on toc.json:
```diff
@@ -268,6 +268,11 @@
+        },
+        {
+          "type": "item",
+          "title": "Safelist IPs",
```
Should we call this page "Webhook IPs" since that's the primary purpose of the page? "Safelisting" is something you might do with those IPs...

mitchellmc replied:
yep

ChezCrawford reviewed (changes requested):
Left a number of questions for review.

mitchellmc commented:
@PagerDuty/lohika-fender & @PagerDuty/lohika-gibson & @PagerDuty/isla FYI!

ChezCrawford commented on docs/webhooks/11-Webhook-IPs.md:
```diff
@@ -0,0 +1,65 @@
+
+The full list of IP addresses that our webhook notifications may come from is:
+
+### US Customers:
```
In other places we refer to these as "Service Regions":

* https://developer.pagerduty.com/docs/ZG9jOjE4NzkyNjM1-public-certificates
* https://support.pagerduty.com/docs/service-regions

ChezCrawford commented on docs/webhooks/11-Webhook-IPs.md:
```diff
@@ -0,0 +1,65 @@
+---
+> Notice: This safelist of IP addresses will take effect April 28th, 2022 - 14:00 UTC
+
+# Webhook Notifications
```
nit: Let's make this the same title as the page: "# Webhook IPs". Stoplight seems to take this and form the URL: `ZG9jOjQ4OTcxMDMx-webhook-notifications` and it would be nice to have that match the sidebar / file name.

ChezCrawford commented on docs/webhooks/11-Webhook-IPs.md:
```diff
@@ -0,0 +1,65 @@
+
+# Webhook Notifications
+
+If you are consuming Generic Webhooks from PagerDuty, use these IPs to make sure make sure they originate from a known PagerDuty Safelist IP.
```
nit: At a minimum, I would remove the term "Generic" since this also includes OWTs. 

I think I would even remove this whole sentence and just kind of say: "This page lists the IP addresses that PagerDuty uses to send webhooks in each of our service regions."

I would move the Notice above below this sentence.

ChezCrawford commented on docs/webhooks/11-Webhook-IPs.md:
```diff
@@ -0,0 +1,65 @@
+
+Until April 28th, 2022 - the [`/webhook_ips`](https://support.pagerduty.com/docs/safelist-ips#webhooks) programmatic endpoint is the authoritative source
+
+After April 28th, 2022 - the `/webhook_ips` endpoint will no longer be updated, and this page should be used for all safelist updates.
```
nit: Lets make `/webhook_ips` a link in both cases or neither case so they look consistent?

ChezCrawford commented:
My final comments here are around the actual list of IP addresses we are going to publish:

*A single IP address should not be used for both our EU and US Service Regions*

As we discussed, it is likely that the IP addresses present in both lists are from [this set of hard-coded OET addresses](https://github.com/PagerDuty/web/blob/main/app/controllers/webhook_ips_controller.rb#L10-L23).

Ideally we should see if we can understand the work that Cloud Infra / Lohika are doing and publish a finalized list.

If that is not possible, then maybe in the short term, we should:

* Remove the hard-coded OET addresses from our EU Service Region list (acknowledging that they will have to go update it after they finish this project).
* Remove the `eu-west-1` IP addresses from our US Service Region list

mitchellmc commented:
> 

So according to this piece of code the US OWT's indeed do not serve from `eu-west-1` anymore, but I'm awaiting final confirmation from Eugene.

I've removed those three IPs from the US Service Region list, and removed all 9 OWT IPs from the EU service region list. 

Ultimately, I hope we can get the new EU OWT NAT Gateway EIPs in this PR, and I'll work with CloudInfra for that; but we should understand that the `/webhook_ips` endpoint is now wrong for both service regions. US has 3 extra IPs that aren't used, and EU has 9.

ChezCrawford commented on docs/webhooks/02-Behavior.md:
```diff
@@ -77,7 +77,7 @@ PagerDuty's [v3 webhooks](../../docs/webhooks/01-Overview.md) are sent with a si
 ### IP Safelists
 
-You can configure your network or server to only accept inbound connections from certain IP addresses. See our [knowledge base article](https://support.pagerduty.com/docs/whitelisting-ips#section-webhooks) for information on obtaining the list of IPs webhooks may come from.  (Previously called IP whitelisting, [see why we changed the name here.](https://twitter.com/secnerdette/status/898314208097 ...
+You can configure your network or server to only accept inbound connections from certain IP addresses. See our [Safelist IPs page](../../docs/webhooks/11-Webhook-IPs.md) for information on obtaining the list of IPs webhooks may come from.  (Previously called IP whitelisting, [see why we changed the name here.](https://twitter.com/secnerdette/status/898314208097427457))
```
nit: "See our [Webhook IPs page]"

ChezCrawford reviewed (approved):
I would change the one tiny comment I made but this looks correct to me.
